src/assembler/encoder.ts

```diff
@@ -296,19 +296,49 @@ function writeDataInstruction(
   }
   let bytes: Byte[] = []
   let alignment: number = 0
-  const values = instruction.options.map((x) => +x)
+  let optionsContainAString: boolean = false
+  let values: number[] = []
+  instruction.options.forEach((x) => {
+    if (x.startsWith('"')) {
+      optionsContainAString = true
+      x = x.substring(0, x.length - 1).substring(1)
+      for (let i = 0; i < x.length; i++) {
+        if (i + 1 < x.length) {
+          // Special case when the string contains an escaped double quote (i.e. "Example'"")
+          if (x.charCodeAt(i) === 39 && x.charCodeAt(i + 1) === 34) {
```

Comment:
as on #53 use charAt and compare with '"' and"'" 
so there are no magic numbers in code

Reply:
Done